R/GeospatialFunctions.R

```diff
@@ -203,44 +230,298 @@ fetchATTAINS <- function(.data) {
+                                                            filter_geom = geospatial_aoi[i],
+                                                            crs = sf::st_crs(geospatial_aoi[i])), silent = TRUE)
+      }
+      
+    }))
+    
+    nhd_catchments <- nhd_catchments %>%
+      purrr::keep(~!is.null(.))
+    
+    try(nhd_catchments <- nhd_catchments %>%
+          dplyr::bind_rows(), silent = TRUE)
+    try(nhd_catchments <- nhd_catchments %>%
+          dplyr::select(nhdplusid,
+                        catchmentareasqkm = areasqkm) %>%
+          dplyr::mutate(NHD.nhdplusid = as.character(nhdplusid),
+                        NHD.resolution = "HR",
+                        NHD.catchmentareasqkm = as.numeric(catchmentareasqkm)) %>%
+          dplyr::select(NHD.nhdplusid, NHD.resolution, NHD.catchmentareasqkm, geometry), silent = TRUE)
+    
+    if(nrow(nhd_catchments) == 0){
+      nhd_catchments <- tibble::tibble(NHD.nhdplusid = character(),
+                                       NHD.resolution = character(),
+                                       NHD.catchmentareasqkm = numeric())
+    }
+    
+    return(nhd_catchments)
+    
+    # If user wants NHDPlus V2...
+  } else if(resolution %in% c("Med", "med")){
+    
+    suppressMessages(suppressWarnings({
+      
+      nhd_catchments <- vector("list", length = nrow(unique_sites))
+      
+      for(i in 1:nrow(unique_sites)){
+        
+        # Use {nhdplusTools} to grab associated catchments...
+        try(nhd_catchments[[i]] <- nhdplusTools::get_nhdplus(AOI = unique_sites[i,], realization = "catchment") %>%
+              dplyr::select(comid = featureid,
+                            catchmentareasqkm = areasqkm) %>%
+              dplyr::mutate(NHD.comid = as.character(comid),
+                            NHD.resolution = "nhdplusV2",
+                            NHD.catchmentareasqkm = as.numeric(catchmentareasqkm)) %>%
+              dplyr::select(NHD.comid, NHD.resolution, NHD.catchmentareasqkm, geometry)
+            , silent = TRUE)
+        
+      }
+      
+      nhd_catchments <- nhd_catchments %>%
+        purrr::keep(~!is.null(.))
+      
+      try(nhd_catchments <- dplyr::bind_rows(nhd_catchments))
+      
+      # if NHD catchments are not in the correct CRS, transform them
+      try(if (sf::st_crs(nhd_catchments) != sf::st_crs(geospatial_data)) {
+        nhd_catchments <- nhd_catchments %>%
+          sf::st_transform(sf::st_crs(geospatial_data)$epsg)
+      }, silent = TRUE)
+      
+    }))
+    
+    if(nrow(nhd_catchments) == 0){
+      nhd_catchments <- tibble::tibble(NHD.comid = character(),
+                                       NHD.resolution = character(),
+                                       NHD.catchmentareasqkm = numeric())
+    }
+    
+    return(nhd_catchments)
+    
+  } else {
+    stop('User-supplied resolution unavailable. Please select between "Med" or "Hi".')
+  }
```

Comment:
This is a great additional function and use of arcgislayers and nhdplusTools; however, I it's called `fetchNHD` and in the description it indicates it returns a list containing all catchment, flowline and waterbody features from either medium res or high res NHD that intersect TADAT WQP locations, but within the function it appears only realization catchment is being called and returned?  I might have missed, but if so would be ideal I'd think to have option to return and dispaly any of the three realizations (i.e. catchment, flowlines or waterbodies)